chainermn/links/n_step_rnn.py

```diff
@@ -22,10 +27,21 @@ def __init__(self, link, communicator, rank_in, rank_out):
         self.rank_in = rank_in
         self.rank_out = rank_out
 
-        if not hasattr(link, 'rnn') or link.rnn not in _rnn_n_cells:
-            raise ValueError('link must be NStepRNN and its inherited link')
-        else:
-            self.n_cells = _rnn_n_cells[link.rnn]
+        if CHAINER_VERSION_OLD_RNN:
+            if not hasattr(link, 'rnn') or link.rnn not in _rnn_n_cells:
+                raise ValueError(
+                    'link must be NStepRNN and its inherited link')
+            else:
+                self.n_cells = _rnn_n_cells[link.rnn]
+
+        else:  # expect Chainer >4.0.0b3
+            check_lstm = issubclass(
+                link.__class__, lconn.n_step_rnn.NStepRNNBase)
```

Comment:
Isn't it more natural to use `isinstance(link, lconn.n_step_rnn.NStepRNNBase)` ?

Reply:
I'm not confident, but I guess it's safer to use `issubclass`.  Do you think we can assume `link` is a direct instance of `NStepRNNBase` ?

Reply:
`isinstance` returns `True` for instances of sub classes. 

```
In [1]: class A(): pass

In [2]: class B(A): pass

In [3]: a = A()

In [4]: b = B()

In [5]: isinstance(a, A)
Out[5]: True

In [6]: isinstance(b, A)
Out[6]: True
```

Reply:
I see, thanks. I will fix this.